### PR TITLE
another way to use shift_labels

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

### CODE REVIEW

File: src/transformers/models/llama/modeling_llama.py [hunk 1]
```diff
@@ -570,9 +570,7 @@ def forward(
         slice_indices = slice(-logits_to_keep, None) if isinstance(logits_to_keep, int) else logits_to_keep
         logits = self.lm_head(hidden_states[:, slice_indices, :])
 
-        loss = None
-        if labels is not None:
-            loss = self.loss_function(logits=logits, labels=labels, vocab_size=self.config.vocab_size, **kwargs)
```
Comment (ydshieh) [inline src/transformers/models/llama/modeling_llama.py:575]: We still need to change in many modeling files, but at least it's more clean than in #37993

Comment (ydshieh) [reply src/transformers/models/llama/modeling_llama.py:575]: @ArthurZucker  if you are happy with this, I will apply the changes to other files.

Comment (ArthurZucker) [reply src/transformers/models/llama/modeling_llama.py:575]: its alright but it creates an indirection! TBH i'd rather we check `labels is not None or "shit_labels" in kwargs`

**Review by ArthurZucker — COMMENTED** [review]
thanks and sorry for the back and forth

**Comment by ydshieh** [comment]
No worry

I will hold on as you mentioned there is a WIP PR #39047
